### PR TITLE
ath79-generic: (re)add support for glinet_6416

### PR DESCRIPTION
**Pull request by AiyionPrime**
@rotanid received the images for testing

- [ ] Must be flashable from vendor firmware
  - [ ] Web interface
  - [ ] TFTP
  - [ ] Other: <specify>
- [X] Must support upgrade mechanism
  - [X] Must have working sysupgrade
    - [X] Must keep/forget configuration (`sysupgrade [-n]`, `firstboot`)
  - [ ] Must have working autoupdate
        *Usually means: Gluon profile name must match image name*
- [X] Reset/WPS/... button must return device into config mode
- [X] Primary MAC address should match address on device label (or packaging)
      (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)
- Wired network
  - [X] should support all network ports on the device
  - [X] must have correct port assignment (WAN/LAN)
    - On devices supplied via PoE, there is usually no explicit WAN/LAN labeling on the hardware.
      The PoE input should be the WAN port in this case.
- Wireless network (if applicable)
  - [X] Association with AP must be possible on all radios
  - [X] Association with 802.11s mesh must work on all radios
  - [X] AP+mesh mode must work in parallel on all radios
- LED mapping
  - Power/sys LED
    - [X] Lit while the device is on
    - [X] Should display config mode blink sequence 
          (https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - Radio LEDs
    - [X] Should map to their respective radio
    - [X] Should show activity
  - Switch port LEDs
    - [X] Should map to their respective port (or switch, if only one led present) 
    - [X] Should show link state and activity
- Outdoor devices only:
  - ~~Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`~~

### CODE REVIEW

**Comment by rotanid** [comment]
@mbaumga tested the device for us.
flashed Freifunk Hannover ar71xx image vH21, sysupgrade to the test-image based on this PR, tested and checked the list in this PR.
didn't test autoupdate or flash from factory.

seems to be merge-ready

**Comment by AiyionPrime** [comment]
Thank you both!

File: targets/ath79-generic [hunk 1]
```diff
@@ -89,6 +89,13 @@ device('enterasys-ws-ap3705', 'enterasys_ws-ap3705i', {
 
 -- GL.iNet
 
+device('gl.inet-6416a-v1', 'glinet_6416', {
```
Comment (NeoRaider) [inline targets/ath79-generic:92]: According to the DTS, the image name should be "gl.inet-6416".

Comment (AiyionPrime) [reply targets/ath79-generic:92]: I really need to read up on the first field. Until now I'm just blindly guessing.
Is the rule just `lowercase of the dts' modelname and replace spaces with dashes`?

Comment (AiyionPrime) [reply targets/ath79-generic]: resolved and rebased to master in edc8d0a433571b84d026320be6a5263ba9b18b45

```diff
-device('gl.inet-6416a-v1', 'glinet_6416', {
+device('gl.inet-6416', 'glinet_6416', {
```

Comment (NeoRaider) [reply targets/ath79-generic]: > I really need to read up on the first field. Until now I'm just blindly guessing. Is the rule just `lowercase of the dts' modelname and replace spaces with dashes`?

The current algorithm is: String is lowercased. All characters that aren't alphanumeric, `.` or `+` are replaced with dashes. Then double dashes and dashes at the end are removed. "-." is replaced with "-". This is implemented here: https://github.com/freifunk-gluon/packages/blob/master/libs/libplatforminfo/src/common.h#L63

Comment (AiyionPrime) [reply targets/ath79-generic]: Thanks, that helped a lot.

**Pull request merged**